sov-modules/sov-modules-api/src/dispatch.rs

```diff
@@ -4,10 +4,12 @@ use sov_state::WorkingSet;
 /// Methods from this trait should be called only once during the rollup deployment.
 pub trait Genesis {
     type Context: Context;
+    type Config;
 
     /// Initializes the state of the rollup.
     fn genesis(
         &self,
+        config: &Self::Config,
```

Comment:
(Question, non-blocking) - why do we take a reference instead of an owned Config here?

Reply:
`init_chain` method does not take `Config` as a parameter (which is fine), so we pass it as `AppTemplate` field. The consequence is that we can't move it (unless we do the trick where we use `Option<Config>` and then we empty it, I can  do it in the next PR).

Reply:
Hmm - maybe we should just add it to `init_chain`

Reply:
Added an issue: https://github.com/Sovereign-Labs/sovereign/issues/165